nixos/lib/make-disk-image.nix

```diff
@@ -302,9 +302,23 @@ in pkgs.vmTools.runInLinuxVM (
         cp ${configFile} /mnt/etc/nixos/configuration.nix
       ''}
 
+
+      ${optionalString (partitionTableType == "efi" || partitionTableType == "hybrid") ''
+        # This is needed for systemd-boot to find ESP from inside runInLinuxVM
+        mkdir /dev/block
+        ln -s /dev/vda1 /dev/block/254:1
+      ''}
+
```

Comment:
I don't think we need to restrict adding the block device symlink only for "EFI" setups. Remove the optionalString. In a way, this brings 

And we probably want to do the `mkdir`/ `ln` up a dozen lines, into the `# Some tools assume these exist` block. After all, some tools assume /dev/block/254:1 exists.

Reply:
I think that these already exist in on-efi boot though. Which is why I'm still confused, it feels like some other component of the system should have been in charge of this setup?

Reply:
This is in the VM that is used to build the derivation. Not on the system that will be running.

The build VM is not booted into a "full" system. Just enough to have a kernel and an environment to run commands as root or requiring kernel module involvement.

Reply:
Ack, yeah, I had some weird false assumption that indeed doesn't make sense given what you pointed out. I agree with the original feedback now.